Add support for more weights and masses

kirinrastogi opened the pull request:
## Units added
* t, Metric Tonne unit of mass.
* slug, Imperial unit of mass.
* N, Newton, metric weight
* short_ton, Short Ton, US unit of weight
* long_ton, Long Ton, Imperial unit of weight

## To consider
I assumed gravity to be 9.80665 m/s<sup>2</sup>
Calculations are now correct.

## Misc
@thegedge This is my first substantial PR to an open source project so I would very much appreciate some feedback, and I'll be quick to fix it on my fork.

part of #84

kirinrastogi commented on test/units/weight_test.rb:
```diff
@@ -90,4 +137,147 @@ class Measured::WeightTest < ActiveSupport::TestCase
+
+  test ".convert_to from t to N" do
+    assert_conversion Measured::Weight, "2000 t", "185014.197509 N"
+  end
```
this is off by a small amount

kirinrastogi commented on test/units/weight_test.rb:
```diff
@@ -90,4 +137,147 @@ class Measured::WeightTest < ActiveSupport::TestCase
+
+  test ".convert_to from N to t" do
+    assert_conversion Measured::Weight, "2000 N", "21.619962434553297 t"
+  end
```
this is off by a larger amount, and I can't figure out why

thegedge commented on lib/measured/units/weight.rb:
```diff
@@ -1,6 +1,10 @@
   unit :kg, value: "1000 g", aliases: [:kilogram, :kilograms]
+  unit :T, value: "1000 kg", aliases: [:tonne, :tonnes]
+  unit :slug, value: "14.593903 kg", aliases: [:slugs]
+  unit :N, value: "9.80665 kg", aliases: [:newtons, :newton]
```
I think this is reversed from what I can tell searching the internet. 1 kilogram is equal to 9.80665 newtons. This may be why you're seeing a discrepancy below (which you've commented on).

kirinrastogi replied:
Nice catch! That makes a lot more sense.

thegedge commented on lib/measured/units/weight.rb:
```diff
@@ -1,6 +1,10 @@
+  unit :T, value: "1000 kg", aliases: [:tonne, :tonnes]
+  unit :slug, value: "14.593903 kg", aliases: [:slugs]
+  unit :N, value: "9.80665 kg", aliases: [:newtons, :newton]
+  unit :t, value: "2000 lb", aliases: [:ton, :tons]
```
My only issue here is that after reading https://en.wikipedia.org/wiki/Short_ton, there is some ambiguity in the term "ton". I think our best bet is just using the long form of "short ton" and "long ton".

Also, https://en.wikipedia.org/wiki/Tonne is telling me that `t` is actually used for the metric ton, or what you have as `tonne` above. It doesn't look like a short ton actually has a shortened (pun unintended!) form.

Finally, perhaps we should add long ton too :)

kirinrastogi replied:
Thanks for the feedback, that is a great idea!

kirinrastogi commented:
@thegedge I added ```long_ton, short_ton``` and remove ```tonne``` as it is just a Megagram that we can alias later with si_unit. I updated the original description of the pr as well.

I wasn't sure how many aliases we wanted to add, we could also add ```longton, longtons```

kirinrastogi commented:
@thegedge I added the aliases you requested, and readded metric tonnes. One thing I noticed is that I am not testing on aliases, does this matter?

Also, the test 
```ruby
test ".convert_to from t to N" do
  assert_conversion Measured::Weight, "2000 t", "19613300 N"
end
```
fails for all versions except 2.4.0, so I am currently looking into that.

kirinrastogi commented:
@thegedge done! Let me know if anything looks off, I had trouble resyncing my fork
